test/cosign_test.py

```diff
@@ -3,48 +3,233 @@
+    return generate_public_key()
+
+
+@pytest.fixture
+def public_key_2():
+    return generate_public_key()
+
+
+def test_json_marshaling_with_annotations(image_digest_tag, image_reference):
     expected_json = '{"critical":{"identity":{"docker-reference":"eu.gcr.io/test/img"},' \
         '"image":{"docker-manifest-digest":' \
-        '"sha256:a904c847d39ae82ec8859ce623ae14ccbfff36d53ce1490b43d9bf5caa47f33b"},' \
-        '"type":"gardener.vnd/oci/cosign-signature"},"optional":{"key":"val"}}'
+        f'"{image_digest_tag}"' \
+        '},"type":"gardener.vnd/oci/cosign-signature"},"optional":{"key":"val"}}'
 
-    image_ref = 'eu.gcr.io/test/img@' \
-        'sha256:a904c847d39ae82ec8859ce623ae14ccbfff36d53ce1490b43d9bf5caa47f33b'
     annotations = {
         "key": "val",
     }
 
     payload = cosign.payload_bytes(
-        image_reference=image_ref,
+        image_reference=image_reference,
         annotations=annotations,
     ).decode('utf-8')
 
     assert payload == expected_json
 
 
-def test_json_marshaling_without_annotations():
+def test_json_marshaling_without_annotations(image_digest_tag, payload):
     expected_json = '{"critical":{"identity":{"docker-reference":"eu.gcr.io/test/img"},' \
         '"image":{"docker-manifest-digest":' \
-        '"sha256:a904c847d39ae82ec8859ce623ae14ccbfff36d53ce1490b43d9bf5caa47f33b"},' \
-        '"type":"gardener.vnd/oci/cosign-signature"},"optional":null}'
-
-    img_ref = 'eu.gcr.io/test/img@' \
-        'sha256:a904c847d39ae82ec8859ce623ae14ccbfff36d53ce1490b43d9bf5caa47f33b'
-
-    payload = cosign.payload_bytes(
-        image_reference=img_ref,
-    ).decode('utf-8')
+        f'"{image_digest_tag}"' \
+        '},"type":"gardener.vnd/oci/cosign-signature"},"optional":null}'
 
-    assert payload == expected_json
+    assert payload.decode('utf-8') == expected_json
 
 
 def test_raise_error_for_img_ref_without_digest():
     img_ref = 'eu.gcr.io/test/img:1.0.0'
     with pytest.raises(ValueError):
         cosign.payload_bytes(image_reference=img_ref)
+
+
+def test_image_signing(image_reference, payload_size, payload_digest, public_key_1, public_key_2):
+    '''
+    Tests signing of an imaginary image multiple times in a row, starting without an existing
+    signature manifest and re-using the created manifest for subsequent signatures (append mode).
+
+    Scenarios:
+    1. Signing using `rsassa-pss` algorithm and `public_key_1`
+    2. Re-signing using same algorithm and key but different signature (as algorithm enforces)
+    3. Re-signing using `rsassa-pkcs1-v1-5` algorithm and `public_key_1`
+    4. Re-signing using `rsassa-pss` algorithm again but with `public_key_2` and changed signature
+    '''
+    oci_client = OciClientMock()
+    signature_image_reference = cosign.default_signature_image_reference(image_reference)
+
+    signature = 'rsassa-pss-signature-dummy'
+    signing_algorithm = model.signing_server.SigningAlgorithm.RSASSA_PSS
+
+    cosign.sign_image(
```

Comment:
seeing the unittest code, I wonder whether we should really split `sign_image` into `sign_manifest` (which expects a manifest-object as input, and returns the signed manifest. we can keep sign_image as a "convenience-function" that will include the retrieval and upload of image-manifest. wdyt?